presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -1101,7 +1102,10 @@ public GroupedExecutionProperties visitMergeJoin(MergeJoinNode node, Void contex
                         left.totalLifespans,
                         left.recoveryEligible && right.recoveryEligible);
             }
-            return GroupedExecutionProperties.notCapable();
+            throw new PrestoException(
+                    INVALID_PLAN_ERROR,
+                    format("When grouped execution can't be enabled, merge join plan is not valid. groupedExecutionEnabled: %s, left node grouped execution capable: %s, " +
+                            "right node grouped execution capable: %s", groupedExecutionEnabled, left.currentNodeCapable, right.currentNodeCapable));
```

Comment:
Let's make it more readable
```java
format("When grouped execution can't be enabled, merge join plan is not valid." +
        "%s is currently set to %s; left node grouped execution capable is %s and " +
        "right node grouped execution capable is %s.",
        GROUPED_EXECUTION,
        groupedExecutionEnabled,
        left.currentNodeCapable,
        right.currentNodeCapable));
```

where `GROUPED_EXECUTION` can be statically imported